lib/tether/ops_linux.go

```diff
@@ -213,7 +213,7 @@ func assignIP(link netlink.Link, endpoint *metadata.NetworkEndpoint) (bool, erro
 	}
 
 	// Set IP address if it's specified - this is now named for for the network role rather than the nic
-	if len(endpoint.Static.IP) > 0 {
+	if endpoint.Static.IP != nil && !endpoint.Static.IP.IsUnspecified() {
```

Comment:
I was having some difficulty with that -- Static would always been non-nil even when the network was using DHCP. I will put it back since it is a safe bet anyway.